packages/entity/src/utils/collections/SerializableKeyMap.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * A serializable object. The serialized value is used to determine equality.
+ */
+export interface ISerializable<TSerialized> {
+  serialize(): TSerialized;
+}
```

Comment:
One thought is this could be:
```suggestion
/**
 * A serializable object. The serialized value is used to determine equality.
 */
export interface ISerializable<TSerialized extends string> {
  toJSON(): TSerialized;
}
```
And then the serializer would call JSON.stringify() and deserializer would call JSON.parse(). However, sometimes we may not want JSON due to its bulkiness compared to potentially more compact formats, in which case the more generic interface makes sense,

Reply:
Precisely, for single field load values it's fine for it to just be the field name and field value. For composite, JSON is likely the most compact. This is only used in-memory (not used as part of any redis keys, for example) so it's changeable in the future too if we need to.